litestar/app.py

```diff
@@ -465,6 +465,9 @@ def __init__(
         self.stores: StoreRegistry = (
             config.stores if isinstance(config.stores, StoreRegistry) else StoreRegistry(config.stores)
         )
+        for _, store in self.stores._stores.items():
+            if not store._should_close_store_manually:
+                self._lifespan_managers.append(store)
```

Comment:
If you have implemented the context manager methods on the store base class.
```suggestion
        for store in self.stores._stores.values():
           self._lifespan_managers.append(store)
```

I'd also move the store registration and this handling logic somewhere within the init before the `self._lifespan_managers` are initialised, so you can simply append it to `config.lifespan`